core/templates/pages/exploration-player-page/services/prediction-algorithm-registry.service.ts

```diff
@@ -43,11 +44,12 @@ export class PredictionAlgorithmRegistryService {
   }
 
   getPredictionService(
-      algorithmId: string, dataSchemaVersion: number): PredictionService {
-    if (this.algorithmIdPredictionServiceMapping.has(algorithmId)) {
+      algorithmId: string, dataSchemaVersion: number):
+       PredictionService | null {
+    if (this.algorithmIdPredictionServiceMapping.get(algorithmId)) {
       const predictionServicesByDataSchemaVersion = (
         this.algorithmIdPredictionServiceMapping.get(algorithmId));
-      if (predictionServicesByDataSchemaVersion.has(dataSchemaVersion)) {
+      if (predictionServicesByDataSchemaVersion.get(dataSchemaVersion)) {
```

Comment:
The code will be a lot cleaner if a variable is assigned to `predictionServicesByDataSchemaVersion.get(dataSchemaVersion)` and is used wherever it is needed.

Reply:
done